### PR TITLE
feat(sync): prevent multiple full syncs with same layerId and geometry type

### LINKED ISSUE #33: prevent multiple full syncs with the same layer id and geometry type
for each pair of `layerId` and `geometryType` there should be only one full sync allowed

### PR DESCRIPTION
**Pull request by melancholiai**
| Question                | Answer                                                                          |
| ---------------- | -------------------------------------------------------------------------- |
| Bug fix         | ✖                                                                        |
| New feature     | ✔                                                                        |
| Breaking change | ✖                                                                        |
| Deprecations    | ✖                                                                        |
| Documentation   | ✖                                                                        |
| Tests added     | ✔                                                                        |
| Chore            | ✖                                                                       |

Closes #33

### CODE REVIEW

File: src/sync/DAL/typeorm/syncRepository.ts [hunk 1]
```diff
@@ -29,4 +29,12 @@ export class SyncRepository extends Repository<DbSync> implements ISyncRepositor
...
     return syncEntity;
   }
+
+  public async findFullSyncByLayerAndGeometry(layerId: number, geometryType: GeometryType): Promise<Sync | undefined> {
+    const fullSyncEntity = await this.findOne({ where: { layerId, geometryType, isFull: true } });
+    if (!fullSyncEntity) {
+      return undefined;
+    }
+    return fullSyncEntity;
+  }
```
Comment (CptSchnitz) [inline src/sync/DAL/typeorm/syncRepository.ts:39]: change it so isFull is also a parameter of the function.

File: src/sync/models/syncManager.ts [hunk 2]
```diff
@@ -25,16 +25,36 @@ export class SyncManager {
...
+
+    const { layerId, geometryType } = updatedSync;
+    if (updatedSync.isFull && (layerId != syncEntity.layerId || geometryType != syncEntity.geometryType)) {
+      const alreadyExistsFullSync = await this.syncRepository.findFullSyncByLayerAndGeometry(layerId, geometryType);
+      if (alreadyExistsFullSync) {
+        throw new FullSyncAlreadyExistsError(
+          `full sync with layer id = ${layerId} and geometry type = ${geometryType} already exists with id ${alreadyExistsFullSync.id}`
+        );
+      }
+    }
```
Comment (CptSchnitz) [inline src/sync/models/syncManager.ts:55]: maybe its better to prevent changing sync type?

**Review by CptSchnitz — CHANGES_REQUESTED** [review]

File: src/sync/models/syncManager.ts (hunk 2 above)
Comment (melancholiai) [reply src/sync/models/syncManager.ts]: done.

File: src/sync/DAL/typeorm/syncRepository.ts (hunk 1 above)
Comment (melancholiai) [reply src/sync/DAL/typeorm/syncRepository.ts]: changed to `findSyncs` which is dynamic to any filter

**Review by CptSchnitz — APPROVED** [review]

**Review by CptSchnitz — APPROVED** [review]

**Pull request merged**